oauth2-server/src/main/java/com/clouway/oauth2/ClientAuthorizationActivity.java

```diff
@@ -49,7 +49,7 @@ public Response execute(String identityId, Request request) {
     }
 
     Set<String> scopes = Sets.newTreeSet(Splitter.on(" ").omitEmptyStrings().split(scope));
-    Optional<Authorization> possibleAuthorizationResponse = clientAuthorizationRepository.authorize(client, identityId, scopes, responseType);
+    Optional<Authorization> possibleAuthorizationResponse = clientAuthorizationRepository.authorize(client, identity.idenityId, identity.scopes, responseType);
```

Comment:
The passed scopes should be from the request, not from the identity as the Client app is requesting them. With this change it doesn't matter what is requesting the client.